### PR TITLE
docs(bodiless-components, vital-meta): Add Vital Meta documentation

### LINKED ISSUE #1678: [DOC] Vital Meta Documentation
## User Story
AS AN end user of the Bodiless documentation,
I WANT to understand how to use the Vital Meta package,
SO THAT I can optimize metadata for my site for use with SEO and Social Sharing.

## Acceptance Criteria
1. The Documentation is updated to include how a Content Editor can use the SEO and Share forms.
2. The Documentation is updated to include how a Site Builder can configure metadata for SEO and Social Sharing.

### PR DESCRIPTION
**Pull request by misterjones**
## Changes

* Add Vital Meta documentation.
* Update existing (Bodiless) Meta documentation.
* Add Vital Meta package to Test Site `package.json`.
  * This is to generate Vital Meta documentation in Test Site.

## Related Issues

Resolves #1678

### CODE REVIEW

File: packages/bodiless-components/doc/Meta.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+          title: 'SEO Data form',
+          description: 'Enter the page level data used for SEO ...',
+        };
+        ```
+01. Then, apply this HOC to the Helmet component:
+    ```js
+    const SeoHelmet = withMetaForm(useMenuOptions, seoFormHeader)(Helmet);
+    ```
+
+### Add Metadata Fields to Editor Interface
```
Comment (hvanyo) [inline packages/bodiless-components/doc/Meta.md:99]:
```suggestion
### Add Metadata Fields to SEO Form in the Editor Interface
```

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md [hunk 2]
```diff
@@ -1,101 +1,53 @@
...
-          description: `Enter the page level data used for SEO ...`
-        };
-        ```
-
-01. Then, apply this HOC to the Helmet component:
-
+01. To add `sitemap.xml` to your `robots.txt` file, set the `ROBOTSTXT_SITEMAP` env variable to your
+    site's `sitemap.xml`:
+    ```shell
+    ROBOTSTXT_SITEMAP='https://www.example.com/sitemap.xml'
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md:16]: This current it is https://example.com/sitemap/sitemap-index.xml

per https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md [hunk 3]
```diff
@@ -1,101 +1,53 @@
...
-      name: 'description',
-      useFormElement: () => useMenuOptionUI().ComponentFormTextArea,
-      label: 'Description',
-      placeholder: 'Rec < 160 char',
-    });
+    plugins: [
+      {
+        resolve: 'gatsby-plugin-canonical-urls',
+        options: {
+          siteUrl: 'https://www.example.com',
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md:37]:
```suggestion
          siteUrl: SITEURL,
```

File: packages/vital-meta/bodiless.docs.json [hunk 4]
```diff
@@ -1,11 +1,16 @@
 {
   "VitalDesignSystem": {
     "Components": {
-      "SEO": {
-        "Metadata": {
-          "vitalMeta.md": "./doc/README.md"
+      "VitalMeta": {
```
Comment (hvanyo) [inline packages/vital-meta/bodiless.docs.json:4]: when I built this I got this...

File: packages/vital-meta/doc/SEO.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+# SEO Data Management
+
+## Default Basic SEO Meta Tags
+
+Vital Meta provides some default basic meta tags to help with the SEO of your site and its pages.
+
```
Comment (hvanyo) [inline packages/vital-meta/doc/SEO.md:6]: If you desire additional meta tags to be added, please follow the Add metadata fields to SEO form in editor interface (/#/Components/Meta?id=add-metadata-fields-to-editor-interface)

File: packages/vital-meta/doc/SEO.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    - If you're on the homepage — and using the Bodiless Schema.org package — you'll be able to
+      enter [organization information](#default-organization-seo-meta-tags).
+
+## Site Builder Details
+
+### Add Metadata Fields to Editor Interface
+
+As Vital Meta is based on [Bodiless Meta](/Components/Meta), the techniques used for adding more
+fields to the Bodiless SEO form can also be applied here.
+
```
Comment (hvanyo) [inline packages/vital-meta/doc/SEO.md:120]: Add in the section about shadowing -- they should do this by via shadowing.. I will get example later this week for you.

File: packages/vital-meta/doc/Share.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+- [Title](#title)
+- [Image](#image)
+- [URL](#url)
+- [Description](#description)
+- [UTM Content](#utm-content)
+- [Twitter Title](#twitter-title)
+- [OG Type](#og-type)
+
+![Social Share Management form](/assets/SocialShareManagementForm.jpg ':size=50%')
+
```
Comment (hvanyo) [inline packages/vital-meta/doc/Share.md:20]: add same blurb here...

If you desire additional open graph tags to be added, please follow the Add metadata fields to SEO form in editor interface (/#/Components/Meta?id=add-metadata-fields-to-editor-interface) and add to the Share Form in the same manner.

File: packages/vital-meta/doc/Share.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    - For details about each meta tag, see the [Default Social Share Meta
+      Tags](#default-social-share-meta-tags) section above.
+
+## Site Builder Details
+
+### Add Metadata Fields to Editor Interface
+
+As Vital Meta is based on [Bodiless Meta](/Components/Meta), the techniques used for adding more
+fields to the Bodiless SEO form can also be applied here.
+
```
Comment (hvanyo) [inline packages/vital-meta/doc/Share.md:126]: Add same bit about doing this via shadowing....
will get an example later this week to add.

File: packages/vital-meta/bodiless.docs.json (hunk 4 above)
Comment (misterjones) [reply packages/vital-meta/bodiless.docs.json:4]: @hvanyo, aha —

- I'm not sure why you're not seeing the "Vital Meta" section from `vital-meta/bodiless.docs.json`...
- But I believe the "SEO > Metadata" sections you're seeing in the ToC are coming from the `bodiless.docs.json` file in **Vital List** — [`vital-list/bodiless.docs.json`](https://github.com/johnsonandjohnson/Bodiless-JS/blob/chore/upgrade-to-gatsby-4-and-tw-3/packages/vital-list/bodiless.docs.json).
  (Which I'm currently working on...)

File: packages/vital-meta/doc/Share.md (hunk 8 above)
Comment (hvanyo) [reply packages/vital-meta/doc/Share.md:130]: we'll try get this into api doc instead of .md files..

File: packages/vital-meta/doc/SEO.md (hunk 6 above)
Comment (hvanyo) [reply packages/vital-meta/doc/SEO.md:123]: we'll try get this into api doc instead of .md files..

**Review by hvanyo — APPROVED** [review]

**Review by dewen — APPROVED** [review]

File: packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md [hunk 9]
```diff
@@ -1,101 +1,70 @@
...
-            label: 'SEO',                    // Button label
-            group: 'page-group',             // Name of a group to which this option belongs
-          },
-        ];
-        ```
-
-    01. `seoFormHeader`: [Optional] Defines SEO form title and description for users.
+    ```shell
+    ROBOTSTXT_SITEMAP='https://www.example.com/sitemap/sitemap-index.xml'
+    ```
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/bodiless-documentation/doc/Development/Guides/BuildingSites/Meta/Metadata.md:20]: ✅  Added explanation of _sitemap_ file versus _sitemap index_ file.

**Review by hvanyo — APPROVED** [review]

**Review by dewen — APPROVED** [review]

**Pull request merged**